src/components/containers/DraggablePanel/DraggablePanel.tsx

```diff
@@ -0,0 +1,181 @@
+import { DragHandle } from "@material-ui/icons";
+import { gray } from "../../../definitions/colors";
+import { CloseCircle } from "../../icons";
+import { primaryFont } from "../../../styleDefinitions/typography";
+
+export type DraggablePanelCoordinatePair = {
+  x: number;
+  y: number;
+};
+
+export type DraggablePanelProps = {
+  /** The contents of the Draggable Panel. */
+  children: ReactNode;
+  /** If set this element will only be able to move within the confines of it's parent elenent. If unset or false, the element will go wherever the user drags it. */
+  confineToParentContainer?: boolean;
+  /** If provided, the panel will live here instead of where they originally live, as defined in the CSS & HTML. */
+  defaultPosition?: DraggablePanelCoordinatePair;
+  /** This meaningful text is used to both render a title and ensure a more WAI-compliant experience.  */
+  panelTitle: string;
+  /** This allows you to specify how tall your panel should be. */
+  initialPanelHeight?: string;
+  /** This allows you to specify how wide your panel should be. */
+  initialPanelWidth?: string;
+  /** This controls weather the panel is visible or not. */
+  isOpen: boolean;
+  /** This allows developers to show or hide the panel title. */
+  showPanelTitle: boolean;
+  /** This event fires when the user's drag completes. */
+  onDragComplete?: (
+    destinationCoordinates: DraggablePanelCoordinatePair
+  ) => void;
+  /** This event fires when the user dismisses a visible panel. */
+  onPanelDismiss?: () => void;
+};
+
+const screenReaderOnly = css`
+  height: 1px;
+  left: -10000px;
+  overflow: hidden;
+  position: absolute;
+  top: auto;
+  width: 1px;
+`;
+
+export function DraggablePanel({
+  confineToParentContainer,
+  children,
+  defaultPosition,
+  initialPanelHeight,
+  initialPanelWidth,
+  isOpen,
+  onDragComplete,
+  onPanelDismiss,
+  panelTitle,
+  showPanelTitle,
+}: DraggablePanelProps) {
+  const [wasDragged, setWasDragged] = useState<boolean>(false);
+  const [isDragging, setIsDragging] = useState<boolean>(false);
+
+  function handleDrag() {
+    !wasDragged && setWasDragged(true);
+  }
+
+  function handleDragStart() {
+    setIsDragging(true);
+  }
+
+  function handleOnDragStop(_: DraggableEvent, data: DraggableData) {
+    setIsDragging(false);
+
+    if (onDragComplete) {
+      onDragComplete({
+        x: data.lastX,
+        y: data.lastY,
+      });
+    }
+  }
+
+  return (
+    <Draggable
+      bounds={confineToParentContainer ? "parent" : false}
+      defaultPosition={defaultPosition || { x: 0, y: 0 }}
+      handle=".dragHandle"
+      onDrag={handleDrag}
+      onStart={handleDragStart}
+      onStop={handleOnDragStop}
+    >
+      <div
+        // As the attribute's name suggests, this helps with automated testing.
+        // At the moment, jsdom and dragging is a bad combo for testing.
+        data-testid={`${panelTitle} ${wasDragged ? "dragged" : "not dragged"}`}
```

Comment:
Is this `data-testid` stripped out for/in production somehow?

Reply:
> Is this `data-testid` stripped out for/in production somehow?

Nah. It's just a [data attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*). You see them in the wild, like this: 
<img width="821" alt="Screen Shot 2023-02-03 at 9 07 41 AM" src="https://user-images.githubusercontent.com/24446573/216637641-3cdc9cd0-7935-4e9b-8744-d0aedc25e8f0.png">.

From testing library, `data-testid` is a workaround for when things get difficult to test programatically:

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)

Reply:
Have you any concerns about the `data-testid`s? As with testing-library and jest, they're pretty industry-standard at this point.

Look at the homepage of chewy.com:

<img width="1202" alt="Screen Shot 2023-02-03 at 3 57 32 PM" src="https://user-images.githubusercontent.com/24446573/216718683-3cc9a0f5-874c-405e-ad71-94c3383ef2a2.png">